api/api/V1/Processing.ts

```diff
@@ -352,10 +367,193 @@ export default function (app: Application, baseUrl: string) {
+   * @apiName PostTracksAndTags
+   * @apiGroup Processing
+   *
+   * Requires super-admin user credentials
+   *
+   * @apiParam {JSON} data Data which defines the tracks and tags (type specific).
+   * @apiParam {Number} AlgorithmId Database ID of the Tracking algorithm details retrieved from
+   * (#FileProcessing:Algorithm) request
+   *
+   * @apiUse V1ResponseSuccess
+   * @apiSuccess {int[]} trackIds of the newly created track.
+   *
+   * @apiUse V1ResponseError
+   *
+   */
+  app.post(
+    `${apiUrl}/:id/tracksAndTags`,
+    extractJwtAuthorisedSuperAdminUser,
+    validateFields([
+      idOf(param("id")),
+      body("data").custom(jsonSchemaOf(ApiMinimalTracksRequestSchema)),
+      idOf(body("algorithmId")),
+    ]),
+    parseJSONField(body("data")),
+    async (request: Request, response: Response, next: NextFunction) => {
+      const recording = await Recording.findByPk(request.params.id);
+      if (!recording) {
+        return next(
+          new AuthorizationError(
+            `Could not find a Recording with an id of '${request.params.id}'`,
+          ),
+        );
+      }
+      const data = response.locals.data;
+      const trackIds: number[] = [];
+      const tracks = [];
+      for (const trackData of data) {
+        const isMasked = await isTrackMasked(
+          recording,
+          trackData,
+          request.body.algorithmId,
+        );
+        if (isMasked) {
+          continue;
+        }
+        const track = prepareTrackToSave(
+          recording,
+          trackData,
+          request.body.algorithmId,
+        );
+        tracks.push(track);
+      }
+      const modelTracks = await Track.bulkCreate(tracks);
+
+      const trackTags = [];
+      const trackTagData = [];
+      for (let i = 0; i < modelTracks.length; i++) {
+        const track = tracks[i];
+        const trackData = data[i];
+        trackIds.push(modelTracks[i].id);
+
+        for (const pred of trackData.predictions) {
+          const modelName = pred.name;
+          const used = modelName === AI_MASTER;
+          let confidence = pred.confidence;
+          // confidence will always be over 50 if it was already put in the 0-100 range
+          if (confidence < 1) {
+            confidence = Math.round(100 * confidence);
+          }
+          const predData = pred as TrackTagData;
+          if (!pred.confident) {
+            predData.raw_tag = pred.tag;
+            pred.tag = "unidentified";
+          }
+
+          const tag = {
+            TrackId: modelTracks[i].id,
+            what: pred.tag,
+            confidence: pred.confidence,
+            automatic: true,
+            model: modelName,
+            UserId: null,
+            used,
+          } as TrackTag;
+          trackTags.push(tag);
+
+          trackTagData.push(predData);
+        }
+
+        delete trackData.predictions;
+        await Track.saveTrackData(track.id, trackData);
```

Comment:
It's probably a bit faster to collect all these promises into an array and then `await` them with `Promise.all` after the loop, so they're saved in parallel as much as possible.